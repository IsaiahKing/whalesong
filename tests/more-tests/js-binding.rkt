#lang planet dyoo/whalesong
(require (planet dyoo/whalesong/js))

(define js-plus
  (js-function->procedure "function(x, y) { return x + y; }"))

(define js-minus
  (js-function->procedure "function(x, y) { return x - y; }"))

(define sleep
  (js-async-function->procedure"function(success, fail, n) { setTimeout(success, n) }"))


"plus: " (js-plus 3 4)
"wait for one second: " (sleep 1000)
"minus:" (js-minus 239748 23)


(for-each (lambda (x)
            (display x)
            (sleep 1000))
          '(hello world testing))


;; I need exception handling...
;;
;(define i-should-fail
;  (js-async-function->procedure  "function(success, fail) { fail('I should fail'); }"))
;(i-should-fail)
