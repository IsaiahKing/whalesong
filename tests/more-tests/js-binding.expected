"plus: "
7
"wait for one second: "
#<undefined>
"minus:"
239725
helloworldtesting